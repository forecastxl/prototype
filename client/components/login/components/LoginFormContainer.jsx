import React, { Component, PropTypes } from 'react'
import { connect } from 'react-redux'
import { actions } from '../../../data/session'
import LoginForm from './LoginForm'

export class LoginFormContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      user: {
        email: '',
        password: ''
      }
    }
    this.changeUser = this.changeUser.bind(this)
    this.processForm = this.processForm.bind(this)
  }

  changeUser(event) {
    this.setState({
      user: {
        ...this.state.user,
        [event.target.name]: event.target.value
      }
    })
  }

  processForm(event) {
    event.preventDefault()
    this.props.requestToken(this.state.user)
  }

  render() {
    return (
      <LoginForm
        errors={this.props.errors}
        onChange={this.changeUser}
        onSubmit={this.processForm}
        user={this.state.user}
      />
    )
  }
}

LoginFormContainer.propTypes = {
  errors: PropTypes.object.isRequired,
  requestToken: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
  errors: state.session.errors
})

const mapActionsToProps = {
  requestToken: actions.requestToken
}

export default connect(mapStateToProps, mapActionsToProps)(LoginFormContainer)
