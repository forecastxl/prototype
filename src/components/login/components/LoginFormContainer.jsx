import React, { Component, PropTypes } from 'react'
import { connect } from 'react-redux'
import { loginUser } from '../actions'
import LoginForm from './LoginForm'

export class LoginFormContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      user: {
        email: '',
        password: ''
      }
    }
    this.changeUser = this.changeUser.bind(this)
    this.processForm = this.processForm.bind(this)
  }

  changeUser(event) {
    this.setState({
      user: {
        ...this.state.user,
        [event.target.name]: event.target.value
      }
    })
  }

  processForm(event) {
    event.preventDefault()
    this.props.loginUser(this.state.user)
  }

  render() {
    return (
      <LoginForm
        onSubmit={this.processForm}
        onChange={this.changeUser}
        user={this.state.user}
      />
    )
  }
}

LoginFormContainer.propTypes = {
  loginUser: PropTypes.func.isRequired
}

export default connect(
  // istanbul ignore next: we only test the unconnected component
  () => ({}), { loginUser })(LoginFormContainer
)
