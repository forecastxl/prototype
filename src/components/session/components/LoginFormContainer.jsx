import React, { Component, PropTypes } from 'react'
import { connect } from 'react-redux'
import { loginUser } from '../actions'
import LoginForm from './LoginForm'
import { NAME } from '../constants'

export class LoginFormContainer extends Component {
  constructor(props) {
    super(props)
    this.state = {
      user: {
        email: '',
        password: ''
      }
    }
    this.changeUser = this.changeUser.bind(this)
    this.processForm = this.processForm.bind(this)
  }

  changeUser(event) {
    this.setState({
      user: {
        ...this.state.user,
        [event.target.name]: event.target.value
      }
    })
  }

  processForm(event) {
    event.preventDefault()
    this.props.loginUser(this.state.user)
  }

  render() {
    return (
      <LoginForm
        errors={this.props.errors}
        onChange={this.changeUser}
        onSubmit={this.processForm}
        user={this.state.user}
      />
    )
  }
}

LoginFormContainer.propTypes = {
  errors: PropTypes.object.isRequired,
  loginUser: PropTypes.func.isRequired
}

const mapStateToProps = state => ({
  errors: state[NAME].errors
})

export default connect(mapStateToProps, { loginUser })(LoginFormContainer)
