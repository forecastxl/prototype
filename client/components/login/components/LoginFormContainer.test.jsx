import React from 'react'
import { shallow, mount } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import { LoginFormContainer } from './LoginFormContainer'

jest.mock('material-ui/TextField')
jest.mock('material-ui/RaisedButton')

describe('<LoginFormContainer />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <LoginFormContainer
        errors={{}}
        requestToken={() => {}}
      />
    )
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('updates user state', () => {
    const wrapper = shallow(
      <LoginFormContainer
        errors={{}}
        requestToken={() => {}}
      />
    )
    const event = {
      target: {
        name: 'password',
        value: 'updated'
      }
    }
    const expected = {
      ...wrapper.state('user'),
      [event.target.name]: event.target.value
    }

    wrapper.instance().changeUser(event)
    expect(wrapper.state('user')).toEqual(expected)
  })

  it('processes logins', () => {
    const wrapper = mount(
      <LoginFormContainer
        errors={{}}
        requestToken={jest.fn()}
      />
    )
    const event = {
      preventDefault: jest.fn()
    }

    wrapper.instance().processForm(event)
    expect(event.preventDefault).toHaveBeenCalled()
    expect(wrapper.prop('requestToken')).toHaveBeenCalledWith(wrapper.state('user'))
  })
})

