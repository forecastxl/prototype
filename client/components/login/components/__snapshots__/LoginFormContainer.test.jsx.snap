exports[`<LoginFormContainer /> renders correctly 1`] = `
<LoginForm
  errors={Object {}}
  onChange={[Function]}
  onSubmit={[Function]}
  user={
    Object {
      "email": "",
      "password": "",
    }
  } />
`;
